rtengine/imagedata.cc

```diff
@@ -1118,9 +1113,7 @@ FramesData::FramesData (const Glib::ustring& fname, std::unique_ptr<RawMetaDataL
 
                 // creating FrameData
                 for (auto currFrame : exifManager.frames) {
-                    FrameData* fd = new FrameData(currFrame, currFrame->getRoot(), roots.at(0));
-
-                    frames.push_back(fd);
+                    frames.push_back(std::unique_ptr<FrameData>(new FrameData(currFrame, currFrame->getRoot(), roots.at(0))));
```

Comment:
Should be `emplace_back(new FrameData...)`.

Reply:
`vector.emplace_back(new ...)` can leak the allocated memory if something throws during resizing the vector (admittedly unlikely here).  Would you still prefer the terser syntax?

As an aside, what is stopping RT from using C++14?  That would let me use `make_unique`.

Reply:
`emplace` vs. `push`: Good point about exception safety. :+1: But if we run out of memory there is nothing that can safe us today (and most likely in future), so that leak won't matter much.

How about reserving `size() + 1` and then emplacing? Best of both worlds. :wink:

Reply:
I'll just push.  I'll make a note near one of the calls.  It'll only be slightly wrong :) and since unique_ptrs are pretty much noexcept, about the only thing that could happen is OOM.

Makes sense about C++14.